### PR TITLE
Native, native swift and react native templates for iOS

### PR DESCRIPTION
**Pull request by wmathurin**
They all use npm + pod to get mobile sdk (and other dependencies)
They all have a template.js script which is invoked by the new forceios to inject app name / company id etc.
The template app can be run directly (after first running install.sh or installios.sh in the case of the react native template).
The react native template is was generated with react-native cli and is compatible with react-native cli. The android portion has not yet been committed (actually it was, but I took it out with https://github.com/forcedotcom/SalesforceMobileSDK-Templates/commit/77e3cf173dab781826f085bf04dbca3314242995).

### CODE REVIEW

**Comment by wmathurin** [comment]
@bhariharan 
Regarding your comments on the original PR:
- I took out the android code from the react native template for now since I have not had a chance to test it - that will come in the android focused PR later (see https://github.com/forcedotcom/SalesforceMobileSDK-Templates/commit/77e3cf173dab781826f085bf04dbca3314242995)
- I updated the project settings for the react native template (see https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/2/commits/b2a17ee2ef74354f65a975ecd3e7a4cdc0ae5079)

**Comment by wmathurin** [comment]
@ivanbogdanov (I don't think you are in the team: https://github.com/orgs/forcedotcom/teams/mobile-sdk)

**Comment by ivanbogdanov** [comment]
Getting this one:


On Nov 1, 2016, at 6:17 PM, Wolfgang Mathurin <notifications@github.com>
wrote:

@ivanbogdanov <https://github.com/ivanbogdanov> (I don't think you are in
the team: https://github.com/orgs/forcedotcom/teams/mobile-sdk)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/2#issuecomment-257746160>,
or mute the thread

File: iOSNativeTemplate/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "iOSNativeTemplate",
+    "version": "0.0.1",
+    "private": true,
+    "dependencies": {
+        "salesforcemobilesdk-ios": "https://github.com/wmathurin/SalesforceMobileSDK-iOS.git#templates"
```
Comment (khawkins) [inline iOSNativeTemplate/package.json]: Should this be changed?

File: ReactNativeTemplate/.flowconfig [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+suppress_type=$FixMe
+
+suppress_comment=\\(.\\|\n\\)*\\$FlowFixMe\\($\\|[^(]\\|(\\(>=0\\.\\(3[0-3]\\|[1-2][0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)
+suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(>=0\\.\\(3[0-3]\\|1[0-9]\\|[1-2][0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)?:? #[0-9]+
+suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
+
+unsafe.enable_getters_and_setters=true
+
+[version]
+^0.33.0
```
Comment (bhariharan) [inline ReactNativeTemplate/.flowconfig]: Shouldn't this be `0.35.0`?

File: ReactNativeTemplate/ios/ReactNativeTemplate/App.entitlements [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>keychain-access-groups</key>
+	<array>
+		<string>$(AppIdentifierPrefix)com.salesforce.iosnativetemplate.iOSNativeTemplate</string>
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/App.entitlements]: I think `(AppIdentifierPrefix)` can be removed here.

File: ReactNativeTemplate/ios/ReactNativeTemplate/AppDelegate.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ Copyright (c) 2015, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/AppDelegate.h]: `-present`.

File: ReactNativeTemplate/ios/ReactNativeTemplate/AppDelegate.m [hunk 5]
```diff
@@ -0,0 +1,236 @@
+/*
+ Copyright (c) 2011-2015, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/AppDelegate.m]: `-present`.

File: ReactNativeTemplate/ios/ReactNativeTemplate/InitialViewController.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
+/*
+ Copyright (c) 2013, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/InitialViewController.h]: `-present`.

File: ReactNativeTemplate/ios/ReactNativeTemplate/InitialViewController.m [hunk 7]
```diff
@@ -0,0 +1,50 @@
+/*
+ Copyright (c) 2013, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/InitialViewController.m]: `-present`.

File: ReactNativeTemplate/ios/ReactNativeTemplate/main.m [hunk 8]
```diff
@@ -0,0 +1,33 @@
+/*
+ Copyright (c) 2015, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplate/main.m]: `-present`.

File: ReactNativeTemplate/ios/ReactNativeTemplateTests/ReactNativeTemplateTests.m [hunk 9]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
```
Comment (bhariharan) [inline ReactNativeTemplate/ios/ReactNativeTemplateTests/ReactNativeTemplateTests.m]: Should this be our copyright?

File: ReactNativeTemplate/package.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "start": "node node_modules/react-native/local-cli/cli.js start",
+    "test": "jest"
+  },
+  "dependencies": {
+    "react": "15.3.2",
+    "react-native": "0.35.0",
+    "salesforcemobilesdk-ios": "https://github.com/wmathurin/SalesforceMobileSDK-iOS.git#templates",
```
Comment (bhariharan) [inline ReactNativeTemplate/package.json]: Change to `forcedotcom`.

File: ReactNativeTemplate/package.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    "react-native-force": "https://github.com/wmathurin/SalesforceMobileSDK-ReactNative.git#unstable"
```
Comment (bhariharan) [inline ReactNativeTemplate/package.json]: Change to `forcedotcom`.

File: iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/App.entitlements [hunk 12]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>keychain-access-groups</key>
+	<array>
+		<string>$(AppIdentifierPrefix)com.salesforce.iosnativeswifttemplate.iOSNativeSwiftTemplate</string>
```
Comment (bhariharan) [inline iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/App.entitlements]: Remove `$(AppIdentifierPrefix)`.

File: iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/InitialViewController.swift [hunk 13]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2014, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/InitialViewController.swift]: `-present`.

File: iOSNativeSwiftTemplate/package.json [hunk 14]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "iOSNativeSwiftTemplate",
+    "version": "0.0.1",
+    "private": true,
+    "dependencies": {
+        "salesforcemobilesdk-ios": "https://github.com/wmathurin/SalesforceMobileSDK-iOS.git#templates"
```
Comment (bhariharan) [inline iOSNativeSwiftTemplate/package.json]: Change to `forcedotcom`.

File: iOSNativeTemplate/iOSNativeTemplate.xcodeproj/xcshareddata/xcschemes/iOSNativeTemplate.xcscheme [hunk 15]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Scheme
+   LastUpgradeVersion = "0800"
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate.xcodeproj/xcshareddata/xcschemes/iOSNativeTemplate.xcscheme]: Change to `0810`.

File: iOSNativeTemplate/iOSNativeTemplate/App.entitlements [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>keychain-access-groups</key>
+	<array>
+		<string>$(AppIdentifierPrefix)com.salesforce.iosnativetemplate.iOSNativeTemplate</string>
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/App.entitlements]: Remove `$(AppIdentifierPrefix)`.

File: iOSNativeTemplate/iOSNativeTemplate/AppDelegate.h [hunk 17]
```diff
@@ -0,0 +1,29 @@
+/*
+ Copyright (c) 2013, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/AppDelegate.h]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/AppDelegate.m [hunk 18]
```diff
@@ -0,0 +1,213 @@
+/*
+ Copyright (c) 2011-2014, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/AppDelegate.m]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/InitialViewController.h [hunk 19]
```diff
@@ -0,0 +1,29 @@
+/*
+ Copyright (c) 2013, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/InitialViewController.h]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/InitialViewController.m [hunk 20]
```diff
@@ -0,0 +1,54 @@
+/*
+ Copyright (c) 2013, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/InitialViewController.m]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/Prefix.pch [hunk 21]
```diff
@@ -0,0 +1,14 @@
+//
+// Prefix header
+//
+
+#import <Availability.h>
+
+#ifndef __IPHONE_4_0
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/Prefix.pch]: We don't need this anymore.

File: iOSNativeTemplate/iOSNativeTemplate/RootViewController.h [hunk 22]
```diff
@@ -0,0 +1,37 @@
+/*
+ Copyright (c) 2011, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/RootViewController.h]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/RootViewController.m [hunk 23]
```diff
@@ -0,0 +1,124 @@
+/*
+ Copyright (c) 2011, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/RootViewController.m]: `-present`.

File: iOSNativeTemplate/iOSNativeTemplate/main.m [hunk 24]
```diff
@@ -0,0 +1,33 @@
+/*
+ Copyright (c) 2014, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline iOSNativeTemplate/iOSNativeTemplate/main.m]: `-present`.

File: iOSNativeTemplate/package.json (hunk 1 above)
Comment (bhariharan) [inline iOSNativeTemplate/package.json]: +1

Comment (wmathurin) [inline iOSNativeTemplate/package.json]: Done here: https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/2/commits/0e9ead0ee9900fdd1d9192d67de877175d6a360c

**Comment by wmathurin** [comment]
Copyright fixes: 
https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/2/commits/0e9ead0ee9900fdd1d9192d67de877175d6a360c
https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/2/commits/1eb7b2762fd7a6685c1dc3bb9d4902c1b1670fab

File: iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/App.entitlements (hunk 12 above)
Comment (wmathurin) [inline iOSNativeSwiftTemplate/iOSNativeSwiftTemplate/App.entitlements]: That's what we have in the sample apps. Are you sure it is not needed?

File: ReactNativeTemplate/.flowconfig (hunk 2 above)
Comment (wmathurin) [inline ReactNativeTemplate/.flowconfig]: It's the version of flow (https://github.com/facebook/flow) not react native.
The latest flow on github is 0.33, while the latest react native is 0.36.

File: ReactNativeTemplate/.buckconfig [hunk 25]
```diff
@@ -0,0 +1,6 @@
+
+[android]
+  target = Google Inc.:Google APIs:23
```
Comment (wmathurin) [inline ReactNativeTemplate/.buckconfig]: I'll look at that one when I do the android work.

**Pull request merged**